controller_manager_tests/test/cm_test.cpp

```diff
@@ -497,6 +496,84 @@ TEST(CMTests, listControllers)
   }
 }
 
+TEST(CMTests, extensibleControllers)
+{
+  ros::NodeHandle nh;
+  ros::ServiceClient load_client   = nh.serviceClient<LoadController>("/controller_manager/load_controller");
+  ros::ServiceClient unload_client = nh.serviceClient<UnloadController>("/controller_manager/unload_controller");
+  ros::ServiceClient switch_client = nh.serviceClient<SwitchController>("/controller_manager/switch_controller");
+  ros::ServiceClient list_client   = nh.serviceClient<ListControllers>("/controller_manager/list_controllers");
+
+  // Load controllers.
+  {
+    LoadController srv;
+    srv.request.name = "extensible_controller";
+    load_client.call(srv);
+    srv.request.name = "derived_controller";
+    load_client.call(srv);
+  }
+
+  // Start the base controller.
+  {
+    SwitchController srv;
+    srv.request.start_controllers.push_back("extensible_controller");
+    srv.request.strictness = srv.request.STRICT;
+    ASSERT_TRUE(switch_client.call(srv));
+    ASSERT_TRUE(srv.response.ok);
+  }
+
+  // Validate the controller is running and claimed one resource.
+  {
+    ListControllers srv;
+    ASSERT_TRUE(list_client.call(srv));
+    ASSERT_EQ(srv.response.controller.size(), 2);
```

Comment:
pedantic: `ASSERT_EQ(2, srv.response.controller.size());` expected value should go first (same happens a little below in this file)

Reply:
I know. I was torn between doing it right vs. matching the rest of this file.